### ISSUE TITLE
initial bert weights take a long time

### ISSUE THREAD

**Issue by yeliu918** [microsoft/unilm#149]
Hi author,
It takes a long time to process model = cls(config, *inputs, **kwargs) in PreTrainedBertModel.from_pretrained(), which aims to Instantiate model. 

Find out the costing-time is the initialization. self.apply(self.init_bert_weights) I don't know why initial bert weights take more than 15 mins while huggingface code take less than 1 min? Since the code is very same. 

Best,
Ye

**Comment by donglixp** [comment]
Hi @yeliu918 Thanks for reporting the initialization issue. Could you provide the command used to reproduce the problem?

**Comment by yeliu918** [comment]
Hi @donglixp Thanks for responding to me. I find the issue there, the version which I use uses the module.weight.data.copy_(torch.Tensor( truncnorm.rvs(-1, 1, size=list(module.weight.data.shape)) * self.config.initializer_range)) in def init_bert_weights(self, module) of class PreTrainedBertModel(nn.Module). That takes a long time. While the normalization           module.weight.data.normal_(mean=0.0, std=self.config.initializer_range) takes a much less time. Could you explain the difference between those two normalization. 
![image](https://user-images.githubusercontent.com/20387632/83053796-e92d3480-a016-11ea-8650-bf1bed71f4af.png)

**Comment by donglixp** [comment]
It's should be caused by a potential bug of numpy. We also noticed that the function `truncnorm.rvs` would be quite slow under some environments (such as different dockers). So we used pytorch's `.normal()` here. There is no significant difference in terms of performance, while `truncnorm.rvs` is more similar to the original BERT's TF implementation.
